### DISCUSSION TITLE
Is it possible to assign a global data file to a layout variable from the page level?

### DISCUSSION THREAD

**Discussion by ptkw** [11ty/eleventy#2690] [Q&A] [answered]
I have a growing number of JSON files kept in a `_data/sources` folder, each similar in structure to the possum example from the documentation:
```
[
  {
    "name": "Fluffy",
    "age": 2
  },
  {
    "name": "Snugglepants",
    "age": 5
  },
  // ...
]
```

My goal is to make a series of posts, each featuring some introductory text and a list generated from one of my data files. Naturally, I would prefer to abstract as much as possible from this process. Ideally, I would like to only provide the name of each data file in each of my posts, and let the layouts do the work.

I made a `list` partial for iterating over data (a JSON file denoted by the `listdata` variable), which for the purposes of simplification can be represented as:
```
<ul>
  {% for item in listdata %}
    <li>the possum named {{ item.name }} is {{ item.age }} years old</li>
  {% endfor %}
</ul>
```

Now, I realize I could already make my post files like this:
```
---
title: 'My possum list'
layout: post
---

Here is some important information concerning the age of some possums:

{% set listdata = sources.possums %}
{%- include "partials/list.njk" -%}
```

In fact, it works well and that's what I'm doing right now. However, this solution strikes me as somewhat clunky and inelegant (a repeatable pattern like this calls for some kind of abstraction instead of copy-pasting the code). I would much prefer to use a `list-post.njk` page template of this form:
```
---
layout: post
---

{{ content | safe }}

{%- include "partials/list.njk" -%}
```
and only set the `listdata` variable in each of the posts. However, I have no idea how to assign a value to `listdata` from this level and whether this is indeed possible.

Putting `{% set listdata = sources.possums %}` in the post body doesn't work (I suppose the assignment happens too late in the generation process).
My attempt to use the post's front matter also failed:
```
---
title: 'My possum list'
layout: list-post
eleventyComputed:
  listdata: {{ sources.possums }}
---
```

I _don't_ want to make my list using a shortcode instead of a partial, because my actual list is much more complicated than the simplified example provided here, and dealing with a long piece of HTML inside a JS string literal without the convenience of Nunjucks would become unmanageable.

I also wouldn't like to merge my data files and store them in keys inside one large JSON (which would enable selecting them simply by a string declared in the front matter).

I would be very grateful for any insight as to whether there is a proper Eleventy way of achieving my goal. If not, I will rest content with manually inserting partials into the post bodies. Thanks for your help anyway!

**Comment by pdehaan** [comment, marked as answer]
You could try passing in a _data/sources filename from your template:

```njk
---
# src/possums.njk
layout: list-post
source: possums
title: Possums
---

<h1>{{ title }}</h1>
```

```njk
---
# src/_includes/layouts/list-post.njk
layout: post
---

{{ content | safe }}

{%- set listdata = sources[source] -%}
{%- include "partials/list.njk" -%}
```

### OUTPUT

```html
<main>
  <h1>Possums</h1>
  <ul>
    <li>the possum named Fluffy is 2 years old</li>
    <li>the possum named Snugglepants is 5 years old</li>
  </ul>
</main>

```

**Reply by ptkw** [reply]
But of course! For some reason it didn't occur to me to use the bracket notation in this context (perhaps because I was mystified by the conversion of the data directory structure into what I thought was some special "dot notation" and not a regular old object with keys). Thank you so much for your help, although I'm sorry it had to consist in pointing out the obvious! (And thanks for all your other thorough and well-formatted answers here, which helped me immensely in learning some finer aspects of Eleventy in the past few days!)
